src/web/components/SharingPermission/BulkAddPermissions.spec.tsx

```diff
@@ -0,0 +1,38 @@
+import { composeStories } from '@storybook/testing-react';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+import * as stories from './BulkAddPermissions.stories';
+
+const { Publisher, HasSharedWithPublisher } = composeStories(stories);
+
+describe('BulkAddPermissions', () => {
+  it('Renders "View Participants" button when there are checkboxes selected', () => {
+    render(<Publisher />);
+    expect(screen.getByRole('button', { name: 'View Participants' })).toBeInTheDocument();
+  });
+  it('Hides "View Participants" button when no checkboxes selected', () => {
+    render(<Publisher />);
+    const dspCheckbox = screen.getByTestId('dsp');
+
+    // uncheck DSP
+    fireEvent.click(dspCheckbox);
+
+    expect(screen.queryByRole('button', { name: 'View Participants' })).not.toBeInTheDocument();
+  });
+  it('Shows warning when removing a shared type', () => {
+    render(<HasSharedWithPublisher />);
+
+    // Expand collapsible
+    fireEvent.click(screen.getByRole('button'));
+    const publisherCheckbox = screen.getByTestId('publisher');
+
+    // uncheck Publisher
+    fireEvent.click(publisherCheckbox);
+
+    expect(
+      screen.getByText(
+        'If you remove the sharing permissions for a participant type, all sharing permissions of that type are removed, including future participants of that type.'
```

Comment:
good question - there difficulty arises in that I think putting the data-testid prop on the <Banner> itself doesn't work. I am guessing this is because the Banner component itself uses `data-testid='banner-icon'`. 

I can put the data-testid on the surrounding div:
```
          <div
            className='remove-recommended-type-warning'
            data-testid='remove-recommended-type-warning'
          >
            <Banner
              type='warning'
              message='If you remove the sharing permissions for a participant type, all sharing permissions of that type are removed, including future participants of that type.'
            />
          </div>
``` 
but then that has the risk of being too closely tied to the DOM than the user experience. 

I sort of don't mind the test breaking if we change the copy, but I can see that maintaining two strings is not ideal. What do you think?

Reply:
Thanks for the explanation, let's just use the copy for testing then